core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -323,35 +343,72 @@ __wasi_errno_t
 wasmtime_ssp_clock_res_get(__wasi_clockid_t clock_id,
                            __wasi_timestamp_t *resolution)
 {
-#ifdef BH_PLATFORM_WINDOWS
-    return __WASI_ENOSYS;
-#else
-    clockid_t nclock_id;
-    if (!convert_clockid(clock_id, &nclock_id))
-        return __WASI_EINVAL;
-    struct timespec ts;
-    if (clock_getres(nclock_id, &ts) < 0)
-        return convert_errno(errno);
-    *resolution = convert_timespec(&ts);
+    if (BH_PLATFORM_WINDOWS) {
+
+        switch (clock_id) {
+            case __WASI_CLOCK_MONOTONIC:
+            {
+                const uint64_t Result =
+                    (uint64_t)(((uint64_t)1000000000) / kFrequency);
+                *resolution = (uint32_t)Result;
+                __wasi_errno_t error = 0;
+                return error;
+            }
+            case __WASI_CLOCK_REALTIME:
+            case __WASI_CLOCK_PROCESS_CPUTIME_ID:
+            case __WASI_CLOCK_THREAD_CPUTIME_ID:
+            {
+                PULONG MaximumTime;
+                PULONG MinimumTime;
+                PULONG CurrentTime;
+                NTSTATUS
+                NTAPI Res;
+                if (Res = NtQueryTimerResolution(&MaximumTime, &MinimumTime,
+                                                 &CurrentTime))
+                    ;
+                unlikely(!NT_SUCCESS_(Res));
+                {
+                    return convert_errno(errno);
```

Comment:
Does `NtQueryTimerResolution` set the `errno` value? Could you point to a resource that defines it?